service/tablet_allocator.cc

```diff
@@ -1084,9 +1077,29 @@ class load_balancer {
                 .avg_tablet_size = avg_tablet_size,
                 .resize_decision = tmap.resize_decision(),
                 .tablet_count = tmap.tablet_count(),
-                .shard_count = shard_count
+                .shard_count = shard_count,
+                .min_tablet_count = 1,
             };
 
+            // FIXME: the table or the replication_strategy might be missing in boost unit tests
+            auto t = _db.get_tables_metadata().get_table_if_exists(table);
+            if (t) {
+                const auto& erm = t->get_effective_replication_map();
+                if (const auto* rs = erm->get_replication_strategy().maybe_as_tablet_aware()) {
+                    size_desc.min_tablet_count = std::max<size_t>(size_desc.min_tablet_count,
+                            rs->calculate_min_tablet_count(t->schema(), erm->get_token_metadata_ptr(), size_desc.target_tablet_size, initial_scale));
+                } else {
+                    auto msg = format("Table {}.{} has no tablet_aware_replication_strategy: uses_tablets={}", t->schema()->ks_name(), t->schema()->cf_name(), erm->get_replication_strategy().uses_tablets());
+                    if (!test_mode) {
+                        on_internal_error(lblogger, msg);
+                    } else {
+                        lblogger.debug("{}", msg);
+                    }
+                }
+            } else if (!test_mode) {
+                on_internal_error(lblogger, format("Table {} does not exist", table));
+            }
```

Comment:
So the tests allow things that cause a crash in real life?

Reply:
yes, unfortunately.